### ISSUE TITLE
Tensor dimension does not match

### ISSUE THREAD

**Issue by emrulhasan-nlp** [almightyGOSU/ANR#4]
Hello, I am trying to run the code. I have successfully run the step 1 and processed the data. However, in step 2, when I run the following command.: python3 PyTorchTEST.py -d "amazon_instant_video" -m "ANRS" -e 10 -p 1 -rs 1337 -gpu 0 -vb 1 -sm "amazon_instant_video_ANRS"

I get and error saying tensor dimensionality miss match, Please see attached. Could you please help me to resolve the issue?

Thanks
![Error](https://user-images.githubusercontent.com/33338872/150042237-dd113224-2430-420f-8b8b-630b2c87c6cf.PNG)

**Comment by almightyGOSU** [comment]
I believe that Line 187 in ModelZoo.py loads the pretrained embeddings.
https://github.com/almightyGOSU/ANR/blob/994562a4ba93ea22eabf1c7c1e3cdfd56c9f7c6a/model/ModelZoo.py#L187

In my experiments, I am using a vocabulary size of 50,000.
Refer to ​Line 33 in PyTorchTEST.py.
https://github.com/almightyGOSU/ANR/blob/994562a4ba93ea22eabf1c7c1e3cdfd56c9f7c6a/PyTorchTEST.py#L33

The size of `self.mdl.wid_wEmbed` is **50,002** x d, as we need to include the `<pad>` and `<unk>` tokens too.
Maybe you are using a different vocabulary size?
E.g. it's something like 37,458 instead?

You can set the vocabulary size using the _-v argument_.

**Comment by emrulhasan-nlp** [comment]
Hi, Thanks for your reply. I set the vocab size to 37458. I still get the error. please see attached. Thanks
![Error2](https://user-images.githubusercontent.com/33338872/150341423-52b53c58-4c85-463d-95c1-b00006f216d9.PNG)

**Comment by almightyGOSU** [comment]
Just set it to **37,456**, and it should work.
The vocabulary size argument doesn't include `<pad>` and `<unk>`.

**Comment by emrulhasan-nlp** [comment]
Thanks,
Now the error shifted to line 183.
![Error3](https://user-images.githubusercontent.com/33338872/150344554-87ea0231-d4bd-4a82-b9ee-9aa122de47cf.PNG)

**Comment by emrulhasan-nlp** [comment]
I set it loss.data instead of loss.data[0]. It's working now. Thank you very much.
